src/hooks/useStoredSearchQuery.ts

```diff
@@ -0,0 +1,13 @@
+import { useState, useEffect } from 'react';
```

Comment:
it can be
```js
import { useState, useEffect, useCallback } from "react";

export const useStoredSearchQuery = (): [string, (query: string) => void] => {
  // Ensures it works in SSR environments
  const getStoredQuery = () => (typeof window !== "undefined" ? localStorage.getItem("searchTerm") || "" : "");

  const [searchQuery, setSearchQuery] = useState<string>(getStoredQuery);

  // Prevents unnecessary rerenders when passed as a dependency
  const updateSearchQuery = useCallback((query: string) => {
    setSearchQuery(query);
    localStorage.setItem("searchTerm", query);
  }, []);

  return [searchQuery, updateSearchQuery];
};

```